knowledge_graph/labelling.py

```diff
@@ -308,3 +309,33 @@ def _assert_datasets_of_the_same_type(*datasets: Dataset):
                 raise ValueError("All datasets must have the same fields")
             if dataset.questions != questions:
                 raise ValueError("All datasets must have the same questions")
+
+
+def label_passages_with_classifier(
+    classifier: Classifier,
+    labelled_passages: list[LabelledPassage],
+    batch_size: int = 16,
+    show_progress: bool = False,
+) -> list[LabelledPassage]:
+    """
+    Label passages using the provided classifier.
+
+    Overwrites any spans that already exist in the labelled passages.
+    """
+
+    input_texts = [lp.text for lp in labelled_passages]
+    model_predicted_spans = classifier.predict(
+        input_texts,
+        batch_size=batch_size,
+        show_progress=show_progress,
+    )
+
+    output_labelled_passages = [
+        labelled_passage.model_copy(
+            update={"spans": model_predicted_spans[idx]},
+            deep=True,
+        )
+        for idx, labelled_passage in enumerate(labelled_passages)
+    ]
+
+    return output_labelled_passages
```

Comment:
i don't think this belongs in this module! 

`labelling.py` is admittedly a bad name - it's supposed to handle the interface between this codebase and  our human labelling system (ie argilla). I'd much prefer to call this module `argilla.py`, but doing so causes all sorts of horrible import issues when importing `argilla` vs `knowledge_graph.argilla` 🫠 

1. if you can think of a clearer name for this file, pls suggest one 😅
2. i still think that this function needs a different home